### PR TITLE
fix(rome_js_analyze): accept const as non camel case

### PR DESCRIPTION
**Pull request by xunilrj**
## Summary

This PR fixes a false flag of all uppercase const as an invalid non-camel case.
Fixes a lot of issues here: https://github.com/rome/tools/runs/8188485278

## Test Plan

```
> cargo test -p rome_js_analyze -- camel_case
```

### CODE REVIEW

**Review by ematipico — COMMENTED** [review]
Could you enable the rule inside `rome.json` file? Doing so it will be a better proof that this PR fixes the false positives

File: crates/rome_js_analyze/src/semantic_analyzers/js/use_camel_case.rs [hunk 1]
```diff
@@ -59,6 +59,21 @@ fn check_is_camel(name: &str) -> Option<State> {
     }
 }
 
+// It is ok to be non camel case when:
+// 1 - is a const variable (eg: const THIS_IS_OK)
```
Comment (ematipico) [inline crates/rome_js_analyze/src/semantic_analyzers/js/use_camel_case.rs:63]:
```suggestion
/// It is OK to be non camel case when:
/// 1 - is a `const` variable (eg: `const THIS_IS_OK`)
```

File: crates/rome_js_analyze/tests/specs/js/useCamelCase.js.snap [hunk 2]
```diff
@@ -34,6 +34,8 @@ console.log({
 let camelCase;
 let longCamelCase;
 
+const THIS_IS_OK = 1;
```
Comment (ematipico) [inline crates/rome_js_analyze/tests/specs/js/useCamelCase.js.snap:37]: Does this fix cover cases like `const { FOO } = env`?

**Review by ematipico — APPROVED** [review]

File: crates/rome_js_analyze/tests/specs/js/useCamelCase.js.snap (hunk 2 above)
Comment (xunilrj) [reply crates/rome_js_analyze/tests/specs/js/useCamelCase.js.snap:41]: Yes. I update the test with more cases.
There is one case we do not deal well.

```
warning[js/useCamelCase]: Prefer variables names in camel case.
   ┌─ useCamelCase.js:31:5
   │
31 │ let UPPER_CASE = 1;
   │     ----------

Safe fix: Rename this symbol to camel case
      | @@ -28,7 +28,7 @@
27 27 |   let camelCase;
28 28 |   let longCamelCase;
29 29 |   
30    | - let UPPER_CASE = 1;
   30 | + let uPPERCASE = 1;
31 31 |   let { UPPER_CASE } = 1;
32 32 |   let [ UPPER_CASE ] = 1;
33 33 |  
```
The problem is that it is easy to fix this, but break other acronyms like IS_HTML -> isHtml.
Not sure the best way here.

File: crates/rome_js_analyze/src/semantic_analyzers/js/use_camel_case.rs (hunk 1 above)
Comment (xunilrj) [reply crates/rome_js_analyze/src/semantic_analyzers/js/use_camel_case.rs]: Done.

**Pull request merged**